aldryn_search/base.py

```diff
@@ -145,3 +153,39 @@ def prepare_fields(self, obj, language, request):
             self.prepared_data['text'] = clean_join(
                 ' ', [prepared_title, prepared_text]
             )
+
+
+class AldrynAppconfigIndexBase(AldrynIndexBase):
+
+    """
+    This class add a check for objects that have app config instances related
+    to it.
+    """
+
+    def check_for_app_config(self, obj):
+        """
+        This should check for app_config in object and if it exists check
+        if app_config namespace exists in some page. If app_config exists
+        and namespace does not exists in pages we should skip indexing for
+        this object. If app_config does not exists for given object or we
+        can't import appconfig stuff, just ignore.
+        """
+        if HAS_APPCONFIG:
+            apphook_field_names = get_apphook_field_names(obj)
+            results = []
+
+            for field_name in apphook_field_names:
+                app_config = getattr(obj, field_name)
+                if app_config:
+                    pages = Page.objects.public().filter(
+                        application_namespace=app_config.namespace
+                    )
+                    results.append(pages.exists())
+            if not any(results):
+                raise DoNotIndex
+
```

Comment:
The above for loop with results[] and later test is rather inefficient (why continue through the loop if you've found something already?). Could you not do something like:

````
for var in loop:
    if condition:
        return
else:
    raise DoNotIndex
````

I think that'd be more pythonic, clearer and efficient.